kv/tikv/storage/server.go

```diff
@@ -0,0 +1,178 @@
+		// The command is readonly or requires access to the DB to determine the keys it will write.
+		reader, err := server.innerServer.Reader(ctxt)
+		if err != nil {
+			return nil, err
+		}
+		txn := kvstore.RoTxn{Reader: reader}
+		resp, latches, err = cmd.Read(&txn)
+		reader.Close()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if latches != nil {
+		// The command will write to the DB.
+		for {
+			wg := server.Latches.AcquireLatches(latches)
+			if wg == nil {
+				break
+			}
+			wg.Wait()
+		}
+		defer server.Latches.ReleaseLatches(latches)
+
+		reader, err := server.innerServer.Reader(ctxt)
+		if err != nil {
+			return nil, err
+		}
+		defer reader.Close()
+
+		// Build an mvcc transaction.
+		txn := kvstore.NewTxn(reader)
+		resp, err = cmd.PrepareWrites(&txn)
+		if err != nil {
+			return nil, err
+		}
+
+		server.Latches.Validate(&txn, latches)
+
+		// Building the transaction succeeded without conflict, write all writes to backing storage.
+		err = server.innerServer.Write(ctxt, txn.Writes)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return resp, nil
+}
+
+// The below functions are Server's gRPC API (implements TikvServer).
+
+// Transactional API.
+func (server *Server) KvGet(ctx context.Context, req *kvrpcpb.GetRequest) (*kvrpcpb.GetResponse, error) {
+	cmd := NewGet(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.GetResponse), err
+}
+
+func (server *Server) KvScan(ctx context.Context, req *kvrpcpb.ScanRequest) (*kvrpcpb.ScanResponse, error) {
+	cmd := NewScan(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.ScanResponse), err
+}
+
+func (server *Server) KvPrewrite(ctx context.Context, req *kvrpcpb.PrewriteRequest) (*kvrpcpb.PrewriteResponse, error) {
+	cmd := NewPrewrite(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.PrewriteResponse), err
+}
+
+func (server *Server) KvCommit(ctx context.Context, req *kvrpcpb.CommitRequest) (*kvrpcpb.CommitResponse, error) {
+	cmd := NewCommit(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.CommitResponse), err
+}
+
+func (server *Server) KvCheckTxnStatus(ctx context.Context, req *kvrpcpb.CheckTxnStatusRequest) (*kvrpcpb.CheckTxnStatusResponse, error) {
+	cmd := NewCheckTxnStatus(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.CheckTxnStatusResponse), err
+}
+
+func (server *Server) KvBatchRollback(ctx context.Context, req *kvrpcpb.BatchRollbackRequest) (*kvrpcpb.BatchRollbackResponse, error) {
+	cmd := NewRollback(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.BatchRollbackResponse), err
+}
+
+func (server *Server) KvResolveLock(ctx context.Context, req *kvrpcpb.ResolveLockRequest) (*kvrpcpb.ResolveLockResponse, error) {
+	cmd := NewResolveLock(req)
+	resp, err := server.Run(&cmd)
+	return resp.(*kvrpcpb.ResolveLockResponse), err
+}
+
+// Raw API.
+func (server *Server) RawGet(ctx context.Context, req *kvrpcpb.RawGetRequest) (*kvrpcpb.RawGetResponse, error) {
+	cmd := NewRawGet(req)
```

Comment:
make raw API back to https://github.com/pingcap-incubator/tinykv/blob/6a6122f5842d7c8b28e66b5f4cfcb82e5181b0fb/kv/tikv/server.go#L134
and remove the related commands

When implementing LAB1, we don't want to involve the term `Latch` and `MvccTxn`